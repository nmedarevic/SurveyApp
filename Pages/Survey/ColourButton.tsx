import * as React from 'react';
import {Button} from 'react-native';
import {TouchableHighlight} from 'react-native-gesture-handler';

export interface ColourButtonProps {
  index?: number;
  selectedIndex?: number;
  name?: string;
  colour?: string;
  onPress?: Function;
}

const buttonSize = 50;
const buttonSizeSelected = 80;

const getButtonStyle = (isSelected: boolean) => ({
  height: isSelected ? buttonSize : buttonSize,
  width: isSelected ? buttonSizeSelected : buttonSize,
  paddingRight: 5,
  paddingLeft: 5,
});
const noop = () => {};
export const ColourButton = ({
  index = 0,
  selectedIndex = 1,
  name = '',
  colour = 'blue',
  onPress = noop,
}: ColourButtonProps) => (
  <TouchableHighlight style={getButtonStyle(index === selectedIndex)}>
    <Button
      key={colour}
      title={`${index + 1}`}
      color={index === selectedIndex ? 'blue' : colour}
      onPress={() => {
        onPress({index, name});
      }}
    />
  </TouchableHighlight>
);
