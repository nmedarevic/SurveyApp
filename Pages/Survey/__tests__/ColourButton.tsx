import React from 'react';
import renderer from 'react-test-renderer';
import {ColourButton} from '../ColourButton';

jest.mock('react-native-gesture-handler', () => ({
  TouchableHighlight: ({children}: any) => ({...children}),
}));
test('renders correctly', () => {
  const tree = renderer.create(<ColourButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with props', () => {
  const props = {
    colour: 'red',
    index: 0,
    selectedIndex: 1,
    name: 'Name',
  };
  const tree = renderer.create(<ColourButton {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with props when selected', () => {
  const props = {
    colour: 'red',
    index: 0,
    selectedIndex: 0,
    name: 'Name',
  };
  const tree = renderer.create(<ColourButton {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
