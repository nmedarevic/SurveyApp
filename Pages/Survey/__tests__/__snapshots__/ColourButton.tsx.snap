// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
      "paddingLeft": 5,
      "paddingRight": 5,
      "width": 50,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button-blue"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "blue",
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with props 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
      "paddingLeft": 5,
      "paddingRight": 5,
      "width": 50,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button-red"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "red",
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with props when selected 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
      "paddingLeft": 5,
      "paddingRight": 5,
      "width": 80,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button-red"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "blue",
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>
</View>
`;
